### ISSUE TITLE
select2 - ajax fetch options

### ISSUE THREAD

**Issue by skaskiewicz** [saltcorn/saltcorn#2928]
**Describe the bug**
when the indicated option is selected and you have a table related via key to another table, in the edit view the selection list does not respond to clicking. Selecting an option is not possible either with the mouse or with the keyboard. does not work when displayed as a single view or when the edit view is opened in a modal

**To Reproduce**
1. Go to: in the edit view there must be an input of type select2 with the option "ajax fetch options" checked
2. Click on: input select2, start searching for content and try to select an item from the result list (list larger than 1 item)
![image](https://github.com/user-attachments/assets/4cbab13b-4b68-4c16-9e9d-5afca1898176)

**Expected behavior**
ability to select items that were searched for when the ajax fetch option is enabled

**Additional context**
saltorn 1.0.0 running in docker with postgresql 15. checked in chrome and firefox

**Comment by christianhugoch** [comment]
Hi, I just prepared the plugin for mobile and didn't see that problem.

Can you create an example app on saltcorn.com (perhaps all with public roles)

**Comment by yoson7** [comment]
I Just want to say that this problem exists regardless ajax, If you are using embedded edit view (sub form) that include select2 field. 

Steps to reproduce:
1. Create Table A for groups (or cities or departments), and table B for related members (like group members, or streets in the city or department members) and table C for common names for members or streets.
2. Create edit view for table B, the edit view includes a select2 input field to select the name from table C.
3. Create edit view for table A and embed the edit view for table B - to create\edit the group and its members.
4. Press the + sign to have the forms of 3 members in the main form of the group.
5. Now you find that you can't edit the names of the members, only the name in the top form.

Why the + to add sub form is duplicating the existing filled form, instead of creating empty fresh sub form?

**Comment by skaskiewicz** [comment]
sorry, I can't upload my project to saltcorn.com correctly. It's too big (nginx returns entity too large error 413). but I've discovered what the problem is. my tables contain data from external sources, which have unique identifiers as unique key (id). tables are imported (discovered) in saltcorn. they are not created in it. however, when I create a table from a csv file (or do it manually and then import), the structure changes relative to the original data. instead of the original unique_number_id field, id is set, and then my two fields are added. the problem is probably the strictly expected database structure with the id field, which is different for me.
steps to reproduce the problem:
create a table with data as below (but not via saltcorn! use psql or your favorite database management program for this). discover (search) the table in saltcorn. let this table be related to another one via its primary key. create a view to edit the parent table with the input select2.

I am attaching the sql code of one table and a screenshot of the relationships between the tables. To reproduce the problem, you only need one table that was not created in saltocrn.

sql code
```
-- public.voivodeship definition

-- Drop table

-- DROP TABLE public.voivodeship;

CREATE TABLE public.voivodeship (
	voivodeship_id varchar(32) NOT NULL,
	voivodeship_name varchar(128) NOT NULL,
	CONSTRAINT voivodeship_pk PRIMARY KEY (voivodeship_id)
);
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('02', 'dolnośląskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('04', 'kujawsko-pomorskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('06', 'lubelskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('08', 'lubuskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('10', 'łódzkie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('12', 'małopolskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('14', 'mazowieckie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('16', 'opolskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('18', 'podkarpackie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('20', 'podlaskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('22', 'pomorskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('24', 'śląskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('26', 'świętokrzyskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('28', 'warmińsko-mazurskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('30', 'wielkopolskie');
INSERT INTO public.voivodeship
(voivodeship_id, voivodeship_name)
VALUES('32', 'zachodniopomorskie');
```

screenshot
![image](https://github.com/user-attachments/assets/f2dace32-1d72-4165-a980-c57f833f7edc)
what the data looks like in the original table (where select2 doesn't work)
![image](https://github.com/user-attachments/assets/99bfeda7-d1dd-4c60-949f-7175a190601d)
what does the data look like in a table created in saltcorn (where select2 works)
![image](https://github.com/user-attachments/assets/94a0bf11-c417-4ece-b29e-4de978f65c1a)

**Comment by christianhugoch** [comment]
thanks, I can re-create it with an external table and ajax

**Comment by christianhugoch** [comment]
Can you try select2 0.5.2, perhaps with a clear cache reload

**Comment by skaskiewicz** [comment]
@christianhugoch 
Now it’s better :)
Select2 allows searching, selecting, and confirming a choice. However, there's still the issue that after selecting a Select2 field, the typing cursor is not automatically placed in the input field (see the image). This requires an extra click to activate the search field.
![Image](https://github.com/user-attachments/assets/d475d93c-667d-48ec-95b0-9acb9f063b6c)

Another concern is the styling: when using the Tabler theme, the background blends with the input field color (see the image). This issue doesn’t appear when displayed in a popup. However, with the SBAdmin2 theme, everything works fine.
tabler
![Image](https://github.com/user-attachments/assets/9f50910e-67e5-4ed2-ad33-52c96660f0bc)
tabler popup
![Image](https://github.com/user-attachments/assets/ef515a1b-745a-4b19-9f6b-ceb57289fdd8)

As for the Selectize plugin, in version 2.5 (the latest version I have available), the search functionality still doesn’t work when the "ajax fetch options" setting is enabled.

**Comment by christianhugoch** [comment]
Okay thanks, I only changed the select2 plugin. I think I can take a look at the color blending within the next few days.

**Comment by christianhugoch** [comment]
Jjust noticed, Tom already did a commit 'improve color on tabler'. It's in 0.5.3 of select2, can you check if this already fixes the color blending?

**Comment by skaskiewicz** [comment]
@christianhugoch 
nope. Still the same. with a slight change - there is a visible frame around the input field.
![Image](https://github.com/user-attachments/assets/43709d29-3c65-450c-9ccd-d72fefdcbef6)
this is what the tabler plugin configuration looks like
![Image](https://github.com/user-attachments/assets/ea72a40a-ad8f-409b-a651-cfeed06b23df)

all select2 fields are visible here
![Image](https://github.com/user-attachments/assets/ca173d20-566b-4fe0-952f-545af61d3408)
here you can see the css class responsible for the color
![Image](https://github.com/user-attachments/assets/adac98de-ff0f-4fe9-80f7-6422fce3532b)
when I remove this class, the input fields start looking normal
![Image](https://github.com/user-attachments/assets/e906462c-9bf2-4f9f-b48a-8e6eb8d3f59a)
single field
![Image](https://github.com/user-attachments/assets/c45768d6-0ca8-4ad8-81a3-28e05a95c549)
view all fields select2
![Image](https://github.com/user-attachments/assets/a9308b9e-7376-4f93-81da-5b1f418652bb)
